src/afio.rs

```diff
@@ -33,6 +33,15 @@ impl AfioExt for AFIO {
     }
 }
 
+/// HAL wrapper around the AFIO registers
+///
+/// Aquired by calling [constrain](trait.AfioExt.html#constrain) on the [AFIO
+/// registers](../pac/struct.AFIO.html)
+///
+/// ```rust
```

Comment:
`rust` is superfluous here as it is automatically assumed, but doc tests for embedded should probably at least feature the `no_run` attribute (if they **can** be compiled but will not run on the host) or even `ignore` if the example cannot compile due to ficticious code.

Reply:
Oh right, I almost forgot doctests are a thing. It would be nice to them where possible, though I feel like it might require a fair amount of extra code in a lot of docstrings.

Reply:
Since it seems to be kind of difficult to run doctests right now, I'd say we should merge this, release a new version to get updated docs on docs.rs, and then sort out doctests later.

Any objections?

Reply:
There's a conflict though.

Reply:
Oh right, there are always changelog problems :) I'll merge #183 then sort these out and merge this